optuna/_imports.py

```diff
@@ -56,10 +66,16 @@ def __exit__(
         """
         if isinstance(exc_value, (ImportError, SyntaxError)):
             if isinstance(exc_value, ImportError):
+                package_name = self._module_to_package_name.get(str(exc_value.name))
+                package_info = (
+                    f"'{exc_value.name}'"
+                    if package_name is None
+                    else f"'{exc_value.name}' (Package Name: {package_name})"
+                )
```

Comment:
IMO, we should not change this package name conversion since this makes a conflict with the existing error message. For example, if we import `google.cloud.storage` and `GCSArtifactStore` in the source code, we face the inconsistent error messages depending the order (importing `google.cloud.storage` first leads to `ModuleNotFoundError: No module named 'google'`, but instantiating `GCSArtifactStore` first leads to `ModuleNotFoundError: No module named 'google-cloud-storage'`). The package name inconsistency is not our responsibility.

Reply:
The following

```python
from optuna.artifacts import GCSArtifactStore
import google.cloud.storage
GCSArtifactStore("a")
```

leads to either:

```
ModuleNotFoundError: No module named 'google'
```

or 

```
ModuleNotFoundError: No module named 'google.cloud.storage'
```

On the other hand, the following:

```python
from optuna.artifacts import GCSArtifactStore
GCSArtifactStore("a")
```

leads to:

```
ImportError: Tried to import 'google' (Package Name: google-cloud-storage) but failed. Please make sure that the package is installed correctly to use this feature. Actual error: No module named 'google'.
```

> but instantiating GCSArtifactStore first leads to ModuleNotFoundError: No module named 'google-cloud-storage'

Answering to this question, we yield both:
- the actual error,
- the original module name (module name is shown as `google` in the example above), and
- the package name.

Although it is true that the package name inconsistency is not our responsibility, the correct installation of the Optuna package and guiding users to the correct configuration are our obligation.

As we make the `google-cloud-storage` an optional dependency, we should notify the correct dependency name if users fail to install it in the first step.